rlclientlib/model_mgmt/restapi_data_transport.cc

```diff
@@ -51,8 +51,8 @@ namespace reinforcement_learning { namespace model_management {
     // Build request URI and start the request.
     auto request_task = _httpcli->request(request).then([&](http_response response) {
       if (response.status_code() != 200){
-        //if the call using HEAD fails with 404 try with GET only once and return the results of GET request call
-        if (response.status_code() == 404 && _retry_get_data){
+        //if the call using HEAD fails try with GET only once and return the results of GET request call

```

Comment:
This is not related to this change. But this is to retry with GET method call once when HEAD method call fails. Model API in front end doesn't have HEAD method call for now, but might be added later. So irrespective of kind of model URI try with HEAD first and if it fails try with GET